clients/vault/src/system.rs

```diff
@@ -940,4 +934,12 @@ impl VaultService {
 		tracing::info!("Got new block at height {startup_height}");
 		Ok(startup_height)
 	}
+
+	/// shuts down the resubmission task running in the background
+	async fn shutdown_wallet(&self) {
+		tracing::info!("shutdown_wallet(): stop the resubmission scheduler");
+		let mut wallet = self.stellar_wallet.write().await;
+		wallet.stop_periodic_resubmission_of_transactions().await;
```

Comment:
I'd also rename the function on the wallet to `try_stop_periodic_resubmission_of_transactions()`